### PR TITLE
Puppet Introduction

### PR DESCRIPTION
**Pull request by hemanik**
Added details to section on Puppet.

### CODE REVIEW

**Comment by nibalizer** [comment]
This looks good. I would recommend squashing your commits into one commit. I think improvements could be some discussion of facter, a description of the communication between agent and master. an example of shipping a configuration file, and the package-file-service pattern.

File: config_management.rst [hunk 1]
```diff
@@ -111,6 +111,81 @@ To prevent documentation drift, see `Ansible documentation site <http://docs.ans
...
+
+Anatomy of a Resource
+---------------------
+
+In Puppet, every resource is an instance of a resource type and is identified by a title, it has a number of attributes (which are defined by the type),
+and each attribute has a value.
+
+Puppet uses its own language to dsecribe and manage resources:
+
+.. code-block:: ruby
```
Comment (igalic) [inline config_management.rst]: puppet works with sphinx

File: config_management.rst [hunk 2]
```diff
@@ -111,6 +111,81 @@ To prevent documentation drift, see `Ansible documentation site <http://docs.ans
...
+An OpenSource software written in Ruby that uses Declarative Domain Specific Language (DSL).
+
+Puppet usually uses an agent/master (client/server) architecture for configuring systems, using the Puppet agent and Puppet master applications.
+It can also run in a self-contained architecture, where each managed server has its own complete copy of your configuration information and
+compiles its own catalog with the Puppet apply application.
+
+Installation
+------------
+
+Puppet has a `installing puppet page <http://docs.puppetlabs.com/puppet/3.8/reference/pre_install.html>`_ on how to get it installed and ready to use,
```
Comment (igalic) [inline config_management.rst]: consider linking to /latest/ or /stable/ rather than 3.8

Comment (hemanik) [inline config_management.rst]: Made suggested changes. Is it good to be merged ?

File: config_management.rst (hunk 1 above)
Comment (hemanik) [inline config_management.rst]: @nibalizer : Thanks, for the suggestions. Added a section on Agent/Master Communication and Package/file/service pattern. Another friend of mine is working on the facter bit for this.

**Pull request merged**